src/App.tsx

```diff
@@ -1,14 +1,59 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterValues } from './types/FilterValues';
 
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [hasLoading, setHasLoading] = useState(false);
+  const [filterByCompleted, setFilterByCompleted] = useState(FilterValues.all);
+  const [filterByTitle, setFilterByTitle] = useState('');
+  const [selectTodo, setSelectTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setHasLoading(true);
+    getTodos()
+      .then(res => {
+        const filteredTodos = res.filter(todo => {
+          switch (filterByCompleted) {
+            case 'all':
+              return true;
+
+            case 'active':
+              return !todo.completed;
+
+            case 'completed':
+              return todo.completed;
+
+            default:
+              return false;
+          }
+        }).filter(todo => {
+          const generalQuery = filterByTitle.toLocaleLowerCase();
+          const generalTitle = todo.title.toLocaleLowerCase();
+
+          return generalTitle.includes(generalQuery);
+        });
+
+        setVisibleTodos(filteredTodos);
+      })
+      .catch(() => {
+        setVisibleTodos([]);
+        throw new Error('Can\'t load Todos with server...')
```

Comment:
it's an error handler it shouldn't throw new errors. It's great to set an empty value for loaded todos, but further we would like to use it for setting an error message in the component state + showing this error message